jersey-servers/src/main/java/com/palantir/remoting3/servers/jersey/AssertionErrorExceptionMapper.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.servers.jersey;
+
+import com.palantir.remoting.api.errors.ErrorType;
+import javax.ws.rs.ext.Provider;
+
+@Provider
+final class AssertionErrorExceptionMapper extends JsonExceptionMapper<AssertionError> {
```

Comment:
It does seem pretty obscure that people are throwing `AssertionErrors` - I've generally only seen them in private constructors (`throw new AssertionError("Not instantiable");`).  However, I think it does makes sense that in the strange case that this happens, we might as well give clients something helpful.

Can we expand it to all Errors? This would get us `NoSuchFieldError`, `NoSuchMethodError` and `NoClassDefFoundError`?

Reply:
Expanding to `Error` or `Throwable` would also cover the occasional `OutOfMemoryError`

Reply:
yeah that was my reticence at the start. I guess there's the thing where we hope that OOMs get handled by the JVM itself?